unitTests/ui/CoreHelpersTest.ts

```diff
@@ -108,17 +108,30 @@ export function CoreHelperTest() {
       );
     });
 
+    it("highlightStreamTextv2 should only highlight a phrase's term once", () => {
```

Comment:
Good test. I would just improve the description to mention that we have duplicates between the terms and phrases. e.g.

```
when the termsToHighlight contains words that are part of phrasesToHighlight,
highlightStreamTextv2 should only highlight a phrase's term once
```

Reply:
Thanks! Done